testing/Core/RecommenderUnitTests.cs

```diff
@@ -0,0 +1,220 @@
+using BalconyBotanica.Core.Algorithm;
+using BalconyBotanica.Core.DomainObjects;
+using BalconyBotanica.Hosts.Models;
+using testing.TestData.Builders;
+
+namespace testing.Core
+{
+    [TestFixture]
+    public class RecommenderUnitTests
+    {
+        // TO DO: all scenario's
+        // [x] happy flow: all answers provided returns plants successful
+        // [ ] happy flow: least amount of anwsers provided returns plants successful
+        // [ ] happy flow: all other answers options [data test?]
+        //  [x]  test sunlight
+        //  [ ]  test balcony size
+        //  [x]  test often water
+        //  [ ]  test functiom
+        //  [ ]  test toxicity
+        // [ ] happy flow: test database items 0, 1, 10 and max amount 
+        // [ ] unhappy flow: null provided returns should-not-be-null-error
+        // [ ] unhappy flow: wrong input returns bad request
+        // [ ] unhappy flow: 
+
+
+        [Test]
+        public void All_Answers_Should_Return_Successfull()
+        {
+            Recommender recommender = new Recommender();
+            RecommendedPlants insertedArray = One.RecommendedPlants;
+            QuizAnswers quizAnswers = One.QuizAnswers;
+
+            var result = recommender.Recommend(insertedArray, quizAnswers);
+
+            Assert.IsInstanceOf<RecommendedPlants>(result);
+            Assert.IsNotNull(result);
+            Assert.IsTrue(result.Plants.Any());
+        }
+
+        // Question: Bij de Recommender werk je met enum-values, juist om het future proof te maken. 
```

Comment:
Question: Bij de Recommender werk je met enum-values, juist om het future proof te maken. 
Moet je de unit tests wel hardcoded specifiek maken, of ook met de enum values werken?

Reply:
I would follow the process of doing boundary testing and testing some standard use cases.

At the boundaries of the `Sunlight` enum we have, `FULL_SHADE` which is at one end of the list and `FULL_SUN` at the other. So you could write tests to check that if the quiz answer is `FULL_SHADE` then only `FULL_SHADE` plants are returned. Then write a test where if the quiz answer is `FULL_SUN` nothing is filtered out. Then you could write one test using one of the middle enum values like `FILTERED_SHADE`.

Do you know about parameterised tests? This makes writing tests where only the input and output differ less repetitive. See https://mcaden.medium.com/writing-parameterized-unit-tests-in-c-run-the-same-test-with-various-inputs-acb6c6e12175
The amount of test code is often much shorter using this technique and adding a new test (if you add more `Sunlight` enum values, for example) is much easier